src/service.ts

```diff
@@ -269,20 +269,15 @@ export class Service {
     output.setData(result);
   }
 
-  static createGist(json: object): Promise<string> {
-    return new Promise((resolve, reject) => {
-      const xhr = new XMLHttpRequest();
-      xhr.addEventListener("load", function() {
-        const jsonURI = JSON.parse(this.response).html_url;
-        resolve(jsonURI);
-      });
-      xhr.addEventListener("error", function() {
-        reject();
-      });
-      xhr.open("POST", "https://api.github.com/gists", true);
-      xhr.setRequestHeader("Content-type", "application/json; charset=utf-8");
-      xhr.send(JSON.stringify(json));
+  static async createGist(json: object): Promise<string> {
+    const url = "https://api.github.com/gists";
+    const response = await fetch(url, {
+      method: "post",
+      body: JSON.stringify(json),
+      headers: new Headers({ "Content-type": "application/json; charset=utf-8" })
     });
+    const jsonres = JSON.parse(await response.text()).html_url;
```

Comment:
rename `jsonres` to `result`. 'jsonres' is not in camel case and not english words, see https://github.com/wasdk/WebAssemblyStudio/wiki/Coding-guidelines#names--identifiers